_episodes/05-accelerating-lammps.md

```diff
@@ -401,67 +432,55 @@ Not surprisingly, the syntax we use is similar to that of **USER-OMP** package:
-> Make sure that the neighbor is built on the CPUs and there is a dynamic load balancing
-> between the CPUs and the GPUs.
+> Employing the full computing workforce to solve your problem may not always be the most
+> profitable. We need to tune this before starting any production run. Derive a command line to
+> submit a LAMMPS job for the LJ system described by the following input file with the following
+> conditions. What command/package keywords should we run so that neighbour list building and force
+> computation are done entirely on the GPUs
+> 
+> * Set `x = y = z = 60`, `t = 500`
+> * Use 2 GPUs and 24 MPI ranks
+> * Neighbour built on CPUs
+> * Dynamic load balancing between CPUs and GPUs dynamic load balancing between the CPUs and the
+>   GPUs.
 >
 > {% capture mycode %}{% include {{ site.snippets }}/ep05/in.lj %}{% endcapture %}
 > {% assign lines_of_code = mycode | strip |newline_to_br | strip_newlines | split: "<br />" %}
+>
 > ```{% for member in lines_of_code %}
 > {{ member }}{% endfor %}
 > ```
 > {: .source}
 > > ## Solution
-> > *** I don't like this, it is currently far too system specific, can we generalise?***
-> > A job submission script is shown below. Note that the number of MPI ranks is fixed
-> > by `#SBATCH --ntasks-per-node=24`. You are requesting for 2 GPUs by using
-> > `#SBATCH --gres=gpu:2`. Rest of the input parameters can be passed to the LAMMPS
-> > executable using command-line switches. The system size can be chosen using
-> > `-v x 60 -v y 60 -v z 60`, length of the run can be decided by `-v t 500`, **GPU**
-> > package and the number of GPUs is chosen by `-pk gpu 2`, the **GPU** package
-> > related fix/pair styles can be chosen using `-sf gpu`, and other **GPU** package
-> > related keywords are chosen using `neigh no newton off split -1.0`. A `no` value of
-> > the `neigh` keyword ensures that the neighbor list is built in the *CPUs*. Dynamic
-> > load balancing option between CPUs and GPUs is chosen using `split -1.0`.
 > >
-> > ~~~
-> > #!/bin/bash -x
-> > #SBATCH --account=ecam
-> > #SBATCH --nodes=1
-> > #SBATCH --ntasks-per-node=24
-> > #SBATCH --output=mpi-out.%j
-> > #SBATCH --error=mpi-err.%j
-> > #SBATCH --time=01:00:00
-> > #SBATCH --partition=gpus
-> > #SBATCH --gres=gpu:2
+> > ```
+> > lmp -v x 60 -v y 60 -v z 60 -v t 500 -sf gpu -pk gpu 2 neigh no newton off split -1.0
+> > ```
+> >
+> > The breakdown of the command is as follows;
 > >
-> > module use /usr/local/software/jureca/OtherStages
-> > module load Stages/Devel-2019a
-> > module load intel-para/2019a
-> > module load LAMMPS/18Feb2020-cuda
+> > * `-v x 60 -v y 60 -v z 60` - Setting system size
+> > * `-v t 500` - Setting length of run
+> > * `-pk gpu 2` - Setting **GPU** package and number of GPUs
+> > * `-sf gpu` - **GPU** package related fix/pair styles
+> > * `neigh no` - A `no` value of `neigh` keyword indicates neighbour list built in the CPUs
+> > * `newton off` - **(FIXME - WHY?)**
```

Comment:
@prithwish-ichec, `newton off`, why? Quick explanation.

Reply:
This is as per LAMMPS manual: "Currently, only an off value is allowed, since all the GPU package pair styles require this setting."  (ref: https://lammps.sandia.gov/doc/package.html)